util/grpcutil/certs.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2021 Andy Dalton
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package grpcutil
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+
+	"google.golang.org/grpc/credentials"
+)
+
+// NewServerTransportCredentials creates an returns a new
+// credentials.TransportCredentials using the given certificate information
+// with a strong TLS server configuration.
+func NewServerTransportCredentials(caCert, cert, key string) (credentials.TransportCredentials, error) {
+	const isServer = true
+	return newTransportCredentials(caCert, cert, key, isServer)
+}
+
+// NewClientTransportCredentials creates an returns a new
+// credentials.TransportCredentials using the given certificate information
+// with a strong TLS client configuration.
+func NewClientTransportCredentials(caCert, cert, key string) (credentials.TransportCredentials, error) {
+	const isServer = false
+	return newTransportCredentials(caCert, cert, key, isServer)
+}
+
+func newTransportCredentials(caCert, cert, key string, isServer bool) (credentials.TransportCredentials, error) {
+	certificate, err := tls.LoadX509KeyPair(cert, key)
+	if err != nil {
+		return nil, err
+	}
+
+	data, err := ioutil.ReadFile(caCert)
+	if err != nil {
+		return nil, err
+	}
+
+	capool := x509.NewCertPool()
+	if !capool.AppendCertsFromPEM(data) {
+		return nil, fmt.Errorf("cannot append ca cert to ca pool")
```

Comment:
```suggestion
		return nil, errors.New("cannot append ca cert to ca pool")
```

(Using a constant string.)